appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockLineChartDataModel.java

```diff
@@ -0,0 +1,108 @@
+
+        // Set the default style properties for the Data Series
+        setDefaultStylingProperties();
+
+        // Adds the Data Series to the Chart.
+        addDataSeriesToChart();
+    }
+
+    @Override
+    public void changeColor(String color) {
+        color = getHexColor(color);
+        dataSeries.setBackgroundColor(color);
+        dataSeries.setPointBackgroundColor(color);
+        dataSeries.setBorderColor(color);
+    }
+
+    @Override
+    public void setElements(String elements) {
+        String[] entries = elements.split(",");
+
+        // Create new list of Data Points
+        ArrayList<DataPoint> dataPoints = new ArrayList<DataPoint>();
+
+        // Since entries come in pairs, we add 2 on each iteration.
+        // Beginning from i = 1 instead of 0 to prevent out of bounds
+        // accesses.
+        for (int i = 1; i < entries.length; i += 2) {
+            try {
+                DataPoint dataPoint = new DataPoint();
+                dataPoint.setX(Double.parseDouble(entries[i-1]));
+                dataPoint.setY(Double.parseDouble(entries[i]));
+                dataPoints.add(dataPoint);
+            } catch (NumberFormatException e) {
+                return; // Wrong input. Do not update entries.
+            }
+        }
+
+        // No data points generated, fallback to default option.
+        if (dataPoints.isEmpty()) {
+            setDefaultElements(dataPoints);
+        } else {
+            // Since we are dealing with a Scatter Data Series, sorting
+            // is a must, because otherwise, the Chart will not look representative.
+            // Consider adding: (1, 2), (5, 3), (2, 5). We want the x = 2
+            // value to be continuous on the Line Chart, rather than
+            // going outside the Chart, which would happen since we
+            // are using a Scatter Chart.
+            dataPoints.sort(Comparator.comparingDouble(DataPoint::getX));
+        }
+
+        // Set the generated data points to the Data Series
+        dataSeries.setDataPoints(dataPoints);
+    }
+
+    @Override
+    protected void setDefaultElements(List<DataPoint> dataPoints) {
+        final int points = 4; // Number of points to add
+
+        // TBD: Might change this in the future.
+        // Generally, this should not cause performance issues because typically there are not
+        // that many data points.
+        Optional<DataPoint> maxYPoint = chartData.getDatasets() // Get all the data sets
+                .stream() // Create a stream
+                .flatMap(l -> ((ScatterDataset)l).getDataPoints().stream()) // Flatten the nested lists to a List of data points
+                .max(Comparator.comparing(DataPoint::getY)); // Get the maximum data point value
+
+        // Get the maximum data point Y value. We take the maximum to ensure
+        // that our newly added default data does not overlap existing lines.
+        double yVal = maxYPoint.map(DataPoint::getY).orElse(0.0);
```

Comment:
The new Line starts from the current maximum value to avoid overlap between data sets (so new default Lines get added on top of all the data, with no overlap). Is this a good approach?